### PR TITLE
Refactor test functions

### PR DESCRIPTION
**Pull request by bpkroth**
Splitting out some work from #557

### CODE REVIEW

File: mlos_bench/mlos_bench/tests/services/remote/ssh/conftest.py [hunk 1]
```diff
@@ -6,118 +6,11 @@
...
-        parent=None,
-    )
+import mlos_bench.tests.services.remote.ssh.fixtures as ssh_fixtures
+
+# Expose those as local names.
+ssh_test_server_hostname = ssh_fixtures.ssh_test_server_hostname
+ssh_test_server = ssh_fixtures.ssh_test_server
+alt_test_server = ssh_fixtures.alt_test_server
+ssh_host_service = ssh_fixtures.ssh_host_service
+ssh_fileshare_service = ssh_fixtures.ssh_fileshare_service
```
Comment (motus) [inline mlos_bench/mlos_bench/tests/services/remote/ssh/conftest.py:16]: why not just
```suggestion
from mlos_bench.tests.services.remote.ssh.fixtures import (
    ssh_test_server_hostname, ssh_test_server, alt_test_server, ssh_host_service, ssh_fileshare_service
)
```

File: mlos_bench/mlos_bench/tests/services/remote/ssh/fixtures.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            service_name=SSH_TEST_SERVER_NAME,
+            hostname=ssh_test_server_hostname,
+            username='root',
+            id_rsa_path=id_rsa_file.name)
+        wait_docker_service_socket(docker_services, ssh_test_server_info.hostname, ssh_test_server_info.get_port())
+        id_rsa_src = f"/{ssh_test_server_info.username}/.ssh/id_rsa"
+        docker_cp_cmd = f"docker compose -p {docker_compose_project_name} cp {SSH_TEST_SERVER_NAME}:{id_rsa_src} {id_rsa_file.name}"
+        cmd = run(docker_cp_cmd.split(), check=True, cwd=os.path.dirname(__file__), capture_output=True, text=True)
+        if cmd.returncode != 0:
+            raise RuntimeError(f"Failed to copy ssh key from {SSH_TEST_SERVER_NAME} container: {str(cmd.stderr)}")
```
Comment (motus) [inline mlos_bench/mlos_bench/tests/services/remote/ssh/fixtures.py:77]: include the return code in the message?

File: mlos_bench/mlos_bench/tests/services/remote/ssh/fixtures.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+@pytest.fixture
+def ssh_host_service(ssh_test_server: SshTestServerInfo) -> SshHostService:
+    """Generic SshHostService fixture."""
+    return SshHostService(
+        config={
+            "ssh_username": ssh_test_server.username,
+            "ssh_priv_key_path": ssh_test_server.id_rsa_path,
+        },
+        global_config={},
+        parent=None,
```
Comment (motus) [inline mlos_bench/mlos_bench/tests/services/remote/ssh/fixtures.py:112]:
```suggestion
```
these are the defaults

File: mlos_bench/mlos_bench/tests/services/remote/ssh/fixtures.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+@pytest.fixture
+def ssh_fileshare_service() -> SshFileShareService:
+    """Generic SshFileShareService fixture."""
+    return SshFileShareService(
+        config={
+            # Left blank to make sure we test per connection overrides.
+        },
+        global_config={},
+        parent=None,
```
Comment (motus) [inline mlos_bench/mlos_bench/tests/services/remote/ssh/fixtures.py:124]:
```suggestion
```
just to make it more concise

**Review by motus — APPROVED** [review]

File: mlos_bench/mlos_bench/tests/services/remote/ssh/conftest.py (hunk 1 above)
Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/services/remote/ssh/conftest.py:16]: This doesn't work unfortunately.  They fixtures need to be local variables to be picked up.

File: mlos_bench/mlos_bench/tests/services/remote/ssh/fixtures.py (hunk 2 above)
Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/services/remote/ssh/fixtures.py]: done

**Pull request merged**